package/MDAnalysis/analysis/contacts.py

```diff
@@ -164,6 +176,199 @@
+    return float(y.sum()) / distances.size
+
+
+def contact_matrix(d, radius, out=None):
+    """Return distance array with True for contacts.
+
+    Parameters
+    ----------
+    d : array
+        is the matrix of distances. The method uses the value of
+        `ContactAnalysis1.radius` to determine if a ``distance < radius``
+        is considered a contact.
+    out: array (optional)
+        If `out` is supplied as a pre-allocated array of the correct
+        shape then it is filled instead of allocating a new one in
+        order to increase performance.
+
+    Returns
+    -------
+    array
+        boolean array of which contacts are formed
+    """
+    if out:
+        out[:] = (d <= radius)
+    else:
+        out = (d <= radius)
+    return out
+
+
+class Contacts(AnalysisBase):
+    """Calculate fraction of native contacts (Q) from a trajectory
+
+    Inputs
+    ------
+    Two string selections for the contacting AtomGroups,
+    the groups could be protein-lipid or protein-protein.
+
+    Use two reference AtomGroups to obtain reference distances (r0)
+    for the cotacts.
+
+    Methods available
+    -----------------
+    Supports either hard-cutoff or soft-cutoff (Best-Hummer like [1]_)
+    contacts.
+
+
+    Parameter choices
+    -----------------
+    There are recommendations and not strict orders.
+    These parameters should be insensitive to small changes.
+    * For all-atom simulations, radius = 4.5 A and lambda_constant = 1.8 (unitless)
+    * For coarse-grained simulations, radius = 6.0 A and lambda_constant = 1.5 (unitless)
+
+    """
+    def __init__(self, u, selection, refgroup, method="cutoff", radius=4.5,
+                 start=None, stop=None, step=None, **kwargs):
+        """
+
+        Parameters
+        ----------
+        u: Universe
+            trajectory
+        selection: tuple(string, string)
+            two contacting groups that change over time
+        refgroup: tuple(AtomGroup, AtomGroup)
+            two contacting groups in their reference conformation
+        radius: float, optional (4.5 Angstroms)
+            radius within which contacts exist
+        method: string
+            either 'cutoff' or 'best-hummer'
+
+        start : int, optional
+            First frame of trajectory to analyse, Default: 0
+        stop : int, optional
+            Last frame of trajectory to analyse, Default: -1
+        step : int, optional
+            Step between frames to analyse, Default: 1
+
+        **kwargs : argument list passed to cut-off function
+            check respective functions for reasonable values
+
+        Attributes
+        ----------
+        results: list
+            Fraction of native contacts for each frame
+        """
+        if method == 'cutoff':
+            self.fraction_contacts = hard_cut
+        elif method == 'best-hummer':
+            self.fraction_contacts = best_hummer_q
+        else:
+            self.fraction_contacts = method
```

Comment:
Yes. I haven't documented much of it yet because I want to wait until others tell me that my usecase examples are good or crap and I have to change the API again.